enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+
+#### General limitations of Placement Groups
+
+- You can create a maximum of 500 placement groups per account in each Region.
+- The name that you specify for a placement group must be unique within your AWS account for the Region.
+- An instance can be launched in one placement group at a time; it cannot span multiple placement groups.
+- You cannot launch AWS Dedicated Hosts in placement groups.
+
+#### Installer changes
+
+We do not plan to add Placement group configuration to the installer machine pool configuration.
+Should users wish to launch the day 0 worker machines within a placement group, they will be able to modify the
+installer generated machineset manifests before the cluster is created.
+We do not intend to support placement groups for control plane machines at all at this point.
+
+We will however need to update the installer cluster deprovisioning logic, as currently it does not know how to remove
+a placement group if one is present within the cluster when the destroy operation is started.
+
+#### Additional permissions
+
+As this enhancement introduces new API calls to the Machine API, we will need to introduce 2 new permissions within the
+CredentialsRequest.
+
+```yaml
+- ec2:DescribePlacementGroups
+- ec2:CreatePlacementGroup
+```
+
+Existing mechanisms should allow these to be added and not cause issues for end users.
+On upgrade, the CCO will block until users of manual mode, have acknowledged that they have included the new
+permissions.
+Users of mint mode will automatically have their IAM roles updated by the CCO.
+
+### Risks and Mitigations
+
+#### Adding new instances to placement groups may cause capacity errors
+
+It is recommended for certain placement group strategies to launch instances using a single launch request. This
+enables AWS to find the appropriate capacity within the datacenter to launch all of the required instances.
+
+OpenShift does not support launching multiple instances at once and as such, we are exposed to adding instances to
+placement groups one by one.
+This increases the likelihood that we will exhaust the capacity within a placement group as we cannot advise AWS ahead
+of time how many instances we need to fit within each placement group.
+
+We currently have no way to mitigate this within Machine API.
+
+If and when capacity limits are reached, Machines will enter a failed phase and an error message will be displayed
+detailing that there was no capacity for the instance to be created.
+Users may delete the Machine and attempt to recreate it later.
+
+If using the Cluster Autoscaler, the Cluster Autoscaler will detect the failed launch and will, after some period (15
+minutes by default), attempt to scale a different group of instances to fulfil the capacity gap, if it is able to.
+
+We recommend that users leverage multiple MachineSets, each with their own Placement Group to minimise the likelihood
```

Comment:
"minimise" how very british. :)  I don't think we ever made a doc declaring american spelling as standard, but now I secretly suspect those people who add spell checkers of trying to backdoor it! 

Entirely tongue in cheek, the spelling is fine :)

Reply:
> I don't think we ever made a doc declaring american spelling as standard, but now I secretly suspect those people who add spell checkers of trying to backdoor it!

If I spell it with a z (pronounced "zed") then I get red squiggly line under it ;) 

I have wondered about whether we had an official spelling recommendation, I know product docs have to be american english but this isn't customer facing (or is it?)

Until I'm forced to write and speak in american english, I'll stick to british and keep pronouncing quay in the british way as well ;)

Reply:
> how very british

The Britishism that trips me up the most is the parsimonious spelling of "fulfil".

> I have wondered about whether we had an official spelling recommendation, I know product docs have to be american english but this isn't customer facing (or is it?)

Upstream has a similar guideline:

> The English-language documentation uses U.S. English spelling and grammar.

https://kubernetes.io/docs/contribute/style/style-guide/

I don't know whether that applies to enhancements though, and anyway, and we're not upstream.  

> Until I'm forced to write and speak in american english, I'll stick to british and keep pronouncing quay in the british way as well ;)

For what it's worth, I think you have parts of the US on your side with that one (I think the British pronunciation is common in the northeastern states).